test/modules/DirectMethodSender/DirectMethodSenderBase.cs

```diff
@@ -54,10 +54,16 @@ public abstract class DirectMethodSenderBase : IDisposable
             catch (Exception e)
             {
                 logger.LogError(e, $"Exception caught with count {this.directMethodCount}");
+                this.directMethodCount++;
```

Comment:
is it easier that initially set directMethodCount to 0 and call "directMethodCount++" before InvokeDeviceMethodAsync.  You need to extract "this.directMethodCount++" out as it calls in both try and catch blocks.